test/litmus/type/date_time_test.exs

```diff
@@ -0,0 +1,66 @@
+defmodule Litmus.Type.DateTimeTest do
+  use ExUnit.Case, async: true
+  doctest Litmus.Type.DateTime
+
+  alias Litmus.Type
+
+  describe "validate_field/3" do
+    test "converts a string to a datetime" do
+      field = "start_date"
+      data = %{"start_date" => "1990-05-01T06:32:00Z"}
+
+      {:ok, expected_date_time, _} = DateTime.from_iso8601(data["start_date"])
+      expected_data = %{"start_date" => expected_date_time}
+
+      type = %Type.DateTime{
+        required: true
+      }
+
+      assert Type.DateTime.validate_field(type, field, data) == {:ok, expected_data}
+    end
+
+    test "does not alter nil" do
+      field = "start_date"
+      data = %{"start_date" => nil}
+
+      type = %Type.DateTime{}
+
+      assert Type.DateTime.validate_field(type, field, data) == {:ok, data}
+    end
+
+    test "errors when required datetime is not provided" do
+      field = "start_date"
+      data = %{}
+
+      type = %Type.DateTime{
+        required: true
+      }
+
+      assert Type.DateTime.validate_field(type, field, data) == {:error, "start_date is required"}
+    end
+
+    test "errors when a the value is not a string" do
+      field = "start_date"
+      data = %{"start_date" => 1234}
+
+      type = %Type.DateTime{
+        required: true
+      }
+
+      assert Type.DateTime.validate_field(type, field, data) ==
+               {:error, "start_date must be a valid ISO-8601 datetime"}
+    end
+
+    test "errors when the value is not in an ISO-8601 format" do
```

Comment:
Let's make it very explicit that we mean ISO-8601 date **and time** format, since technically that's a valid ISO-8601 date.